botorch/fit.py

```diff
@@ -267,6 +278,56 @@ def _fit_fallback(
     raise ModelFittingError("All attempts to fit the model have failed.")
 
 
+@dispatcher.register(_ApproximateMarginalLogLikelihood, object, object)
+def _fit_fallback_approximate(
+    mll: _ApproximateMarginalLogLikelihood,
+    _: Type[Likelihood],
+    __: Type[ApproximateGPyTorchModel],
+    *,
+    closure: Optional[TLossClosure] = None,
+    data_loader: Optional[Union[DataLoader, Dict[str, Any]]] = None,
+    optimizer: Optional[Callable] = None,
+    full_batch_cutoff: int = 1024,
+    **kwargs: Any,
+) -> _ApproximateMarginalLogLikelihood:
+    r"""Fallback method for fitting approximate Gaussian processes.
+
+    Args:
+        closure: Loss closure for `optimizer`.
+        optimizer: The underlying optimization algorithm to run. Default to
+            `fit_gpytorch_scipy` when `closure=None` and the model's internal
+            training set has no more than `full_batch_cutoff` observations; otherwise,
+            defaults to `fit_gpytorch_torch`.
+        data_loader: Convience keyword for passing in a DataLoader instance or dict of
+            keyword arguments passed to `get_data_loader` to obtain one. May only be
```

Comment:
It feels a bit weird that `data_loader` can either be an instance or a set of kwargs passed to a factory function.